packages/next/src/client/components/router-reducer/reducers/server-action-reducer.ts

```diff
@@ -57,12 +57,12 @@ async function fetchServerAction(
 ): Promise<FetchServerActionResult> {
   const body = await encodeReply(actionArgs)
 
-  const newNextUrl = extractPathFromFlightRouterState(state.tree)
-  // only pass along the `nextUrl` param (used for interception routes) if it exists and
-  // if it's different from the current `nextUrl`. This indicates the route has already been intercepted,
-  // and so the action should be as well. Otherwise the server action might be intercepted
-  // with the wrong action id (ie, one that corresponds with the intercepted route)
-  const includeNextUrl = state.nextUrl && state.nextUrl !== newNextUrl
+  // only pass along the `nextUrl` param (used for interception routes) if the current route was intercepted.
+  // If the route has been intercepted, the action should be as well.
+  // Otherwise the server action might be intercepted with the wrong action id
+  // (ie, one that corresponds with the intercepted route)
+  const includeNextUrl =
+    state.nextUrl && hasInterceptionRouteInCurrentTree(state.tree)
```

Comment:
The previous implementation here solved a similar problem, but it relied on an incorrect assumption about `nextUrl`. This more accurately determines if an interception marker is present in the tree. 

Confirmed that the previous test that was added for this case still passes ([test ref](https://github.com/vercel/next.js/blob/6687634ab849477855e5130859ccb60edcf291a3/test/e2e/app-dir/actions/app-action.test.ts#L1011-L1049))